addons/stock/wizard/stock_replenishment_info.py

```diff
@@ -73,38 +95,98 @@ def _format_description(description):
                 'virtual': orderpoint.trigger == 'manual' and orderpoint.create_uid.id == SUPERUSER_ID,
             })
 
-    @api.depends('orderpoint_id')
-    def _compute_json_replenishment_history(self):
-        for replenishment_report in self:
-            replenishment_history = []
+    def _get_period_of_time(self):
+        self.ensure_one()
+        start_date = fields.Datetime.now()
+        limit_date = fields.Datetime.now()
+        if self.based_on == 'one_week':
+            start_date = start_date - relativedelta(weeks=1)
+        elif self.based_on == 'one_month':
+            start_date = start_date - relativedelta(months=1)
+        elif self.based_on == 'three_months':
+            start_date = start_date - relativedelta(months=3)
+        elif self.based_on == 'one_year':
+            start_date = start_date - relativedelta(years=1)
+        else:  # Relative period of time.
             today = fields.Datetime.now()
-            first_month = subtract(today, months=2)
-            date_from, dummy = get_month(first_month)
-            dummy, date_to = get_month(today)
+            start_date = datetime(year=today.year - 1, month=today.month, day=1)
+
+            if self.based_on == 'last_year_2':
+                start_date += relativedelta(months=1)
+            elif self.based_on == 'last_year_3':
+                start_date += relativedelta(months=2)
+
+            if self.based_on == 'last_year_quarter':
+                limit_date = start_date + relativedelta(months=3)
+            else:
+                limit_date = start_date + relativedelta(months=1)
+        return start_date, limit_date
+
+    def _prepare_graph_data(self, daily_demand=0):
+        self.ensure_one()
+        if not daily_demand:
+            ordering_period = 0
+            axe_x_vals = ['', ' ']
```

Comment:
`axe_x_vals` => `x_axis_vals`

one of those weird English exceptions: axis = singular, axes = plural, axe = :axe: